clowdapp.yaml

```diff
@@ -8,6 +8,8 @@ objects:
   kind: ClowdApp
   metadata:
     name: "${APP_NAME}"
+    annotations:
+      ignore-check.kube-linter.io/no-readiness-probe
```

Comment:
@vkrizan @skateman After looking at [this article](https://www.civo.com/learn/yaml-best-practices-using-kubelinter#ignoring-a-check) closer, I think that the `annotations` need to be under `metadata` like this. I'm not sure how to only ignore the checks for the jobs and not the deployments. Maybe I can add `metadata:` under `jobs:` and then put this annotation in there? 

The article that I attached here is the one that AppSRE sent me, by the way.

Reply:
This is starting to be tricky with the ClowdApp object. This annotation would (likely) suppress it for whole ClowdApp. Unfortunately there are no metadata nor annotation parameters on [Clowder's Job](https://github.com/RedHatInsights/clowder/blob/41827e006117220fb7e10ed04bd57343ebf3fadf/apis/cloud.redhat.com/v1alpha1/clowdapp_types.go#L76-L115) and neither on [Clowder's PodSpec](https://github.com/RedHatInsights/clowder/blob/41827e006117220fb7e10ed04bd57343ebf3fadf/apis/cloud.redhat.com/v1alpha1/clowdapp_types.go#L206-L248). 

This leaves us four options:
1) leave it as you written it - which might not be acceptable,
2) write some readiness probe,
3) don't do any changes and ignore the issue,
4) ask platform to add metadata&annotations to `cloud.redhat.com/v1alpha1/PodSpec`.

I'd like to understand where this problem is coming from. Whether AppSRE has some guidance that a template has to comply with some linter. I'll ask them on the Jira.

Reply:
Right, I think it will correctly disable the check if we leave it how I have it but like you said, it will probably suppress it for everything. Maybe that could be okay, at least as a short term solution, since we know that we have the checks passing for everything else besides these few failing checks?
I also have a [separate ticket open](https://github.com/RedHatInsights/compliance-backend/pull/1120) for a similar issue where I need to suppress the min replica check for `compliance-ssg-service` and I don't know what a good workaround for that one would be besides disabling the alert like this

Reply:
The linter warning suppression for everything might be okay. Try reaching out to AppSRE to get it confirmed. Maybe they have their own policy.

For the replicas, the workaround can be to have three replicas. It would likely take some of namespace resources, but it would resolve the warning.